dbeam/src/test/scala/com/spotify/dbeam/PsqlAvroJobTest.scala

```diff
@@ -85,19 +85,17 @@ class PsqlAvroJobTest extends FlatSpec with Matchers with BeforeAndAfterAll {
   it should "validate replication" in {
     val partition = new DateTime(2027, 7, 31, 0, 0, DateTimeZone.UTC)
     val lastReplication = new DateTime(2027, 8, 1, 0, 0, DateTimeZone.UTC)
+    val partitionPeriod = Days.ONE
 
-    val actual = PsqlAvroJob.validateReplication(partition, lastReplication)
-
-    actual should be (partition)
+    PsqlAvroJob.isReplicationDelayed(partition, lastReplication, partitionPeriod) shouldBe false
   }
 
-  ignore should "fail on too replication" in {
+  it should "fail on delayed replication" in {
     val partition = new DateTime(2027, 7, 31, 0, 0, DateTimeZone.UTC)
-    val lastReplication = new DateTime(2027, 7, 19, 0, 0, DateTimeZone.UTC)
+    val lastReplication = new DateTime(2027, 7, 31, 10, 0, DateTimeZone.UTC)
+    val partitionPeriod = Days.ONE
 
-    a[IllegalArgumentException] should be thrownBy {
-      PsqlAvroJob.validateReplication(partition, lastReplication)
-    }
+    PsqlAvroJob.isReplicationDelayed(partition, lastReplication, partitionPeriod) shouldBe true
   }
```

Comment:
I think we can have a few more tests here, to validate the boundaries.

Like:
- `x<y`
- `x=y`
- `x>y`

And maybe also:
- `x-1<y`
- `x<y+1`